src/main/java/com/toofifty/easyblastfurnace/state/InventoryState.java

```diff
@@ -68,13 +70,20 @@ public boolean has(int ...itemIds) {
         return getQuantity(itemIds) > 0;
     }
 
-    public int getFreeSlots()
+    public int getFreeSlots(boolean ignoreBarsAndOres)
     {
         load();
 
         int freeSlots = 28;
+        int[] barsAndOres = new int[]{
+            ItemID.IRON_BAR, ItemID.MITHRIL_BAR, ItemID.ADAMANTITE_BAR, ItemID.RUNITE_BAR, ItemID.GOLD_BAR, ItemID.STEEL_BAR,
+            ItemID.IRON_ORE, ItemID.MITHRIL_ORE, ItemID.ADAMANTITE_ORE, ItemID.RUNITE_ORE, ItemID.GOLD_ORE, ItemID.COAL
+        };
+
         for (Item item : inventory.getItems()) {
-            if (item.getQuantity() > 0) {
+            if (ignoreBarsAndOres && IntStream.of(barsAndOres).noneMatch(id -> id == item.getId()) && item.getQuantity() > 0) {
+                freeSlots--;
```

Comment:
Is this _really_ still the amount of free slots? I feel like the new calculation should be it's own method

Reply:
Fair, I've made a separate method `getFreeSlotsIncludingOresAndBars` to address this.